retrievalmarket/impl/provider.go

```diff
@@ -275,6 +276,42 @@ func (p *Provider) ListDeals() map[retrievalmarket.ProviderDealIdentifier]retrie
 	return dealMap
 }
 
+// HandleAskStream is called when the provider receives a request for the
+// retrieval ask (how much the provider charges for retrieval)
+func (p *Provider) HandleAskStream(s rmnet.RetrievalAskStream) {
```

Comment:
@dirkmc  This wont work. With Dynamic pricing, we quote a different price for each deal based on a number of factors. Just returning the ask from the ask store wont work. Maybe the storage provider has an external script configured for example. You want the client to give you the params it gives you when it actually makes the deal (pieceCid etc.) in the request for this stream and then run it through the same pricing function that we run during retrieval.